src/compiler/emitter.ts

```diff
@@ -4046,7 +4006,7 @@ module ts {
                 var preambleEmitted = writer.getTextPos() !== outPos;
 
                 if (!preambleEmitted && nodeEndIsOnSameLineAsNodeStart(body, body)) {
-                    for (var i = 0, n = body.statements.length; i < n; i++) {
+                    for (var i = startIndex, n = body.statements.length; i < n; i++) {

```

Comment:
No.  Functions with preambles will alwys be on multiple lines.  If thre is no preamble, then the function will stay on a single line if it was written on a single line.  Otherwise, it will become multiple lines.